File: sdks/js/samples/VideoHeartbeatSample/script/app/analytics.provider.js
(function(ADB, Configuration, SamplePlayerDelegate) {
    'use strict';

    var NotificationCenter = ADB.core.NotificationCenter;
    var VideoHeartbeat = ADB.va.VideoHeartbeat;
    var ConfigData = ADB.va.ConfigData;

    function AnalyticsProvider(appMeasurement) {
        this._appMeasurement = appMeasurement;

        this._player = null;
        this._videoHeartbeat = null;
        this._playerDelegate = null;
    }

    AnalyticsProvider.prototype.attachToPlayer = function(player) {
        if (this._player) {
            this.detachFromPlayer();
        }

        this._player = player;

        // We register as observers to various VideoPlayer events.
        NotificationCenter().addEventListener(PlayerEvent.VIDEO_LOAD, this._onLoad, this);
        NotificationCenter().addEventListener(PlayerEvent.VIDEO_UNLOAD, this._onUnload, this);
        NotificationCenter().addEventListener(PlayerEvent.PLAY, this._onPlay, this);
        NotificationCenter().addEventListener(PlayerEvent.PAUSE, this._onPause, this);
        NotificationCenter().addEventListener(PlayerEvent.SEEK_START, this._onSeekStart, this);
        NotificationCenter().addEventListener(PlayerEvent.SEEK_COMPLETE, this._onSeekComplete, this);
        NotificationCenter().addEventListener(PlayerEvent.COMPLETE, this._onComplete, this);

        this._playerDelegate  = new SamplePlayerDelegate(this._player);

        var configData = new ConfigData(Configuration.HEARTBEAT.TRACKING_SERVER,
            Configuration.HEARTBEAT.JOB_ID,
            Configuration.HEARTBEAT.PUBLISHER);

        configData.ovp = Configuration.HEARTBEAT.OVP;
        configData.sdk = Configuration.HEARTBEAT.SDK;
        configData.channel = Configuration.HEARTBEAT.CHANNEL;

        // Comment or explicitly set this to false for production sites.
        configData.debugLogging = true;

        this._videoHeartbeat = new VideoHeartbeat(this._appMeasurement, this._playerDelegate);
        this._videoHeartbeat.configure(configData);
    };

    AnalyticsProvider.prototype.detachFromPlayer = function() {
        if (this._player) {
            this._videoHeartbeat.destroy();
            this._videoHeartbeat = null;
            this._playerDelegate = null;

            NotificationCenter().removeAllListeners(this);
            this._player = null;
        }
    };

    AnalyticsProvider.prototype._onLoad = function(e) {
        console.log('Player event: VIDEO_LOAD');
        this._videoHeartbeat.trackVideoLoad();
    };

    AnalyticsProvider.prototype._onUnload = function(e) {
        console.log('Player event: VIDEO_UNLOAD');
        this._videoHeartbeat.trackVideoUnload();
    };

    AnalyticsProvider.prototype._onPlay = function(e) {
        console.log('Player event: PLAY');
        this._videoHeartbeat.trackPlay();
    };

    AnalyticsProvider.prototype._onPause = function(e) {
        console.log('Player event: PAUSE');
        this._videoHeartbeat.trackPause();
    };

    AnalyticsProvider.prototype._onSeekStart = function(e) {
        console.log('Player event: SEEK_START');
        this._videoHeartbeat.trackSeekStart();
    };

    AnalyticsProvider.prototype._onSeekComplete = function(e) {
        console.log('Player event: SEEK_COMPLETE');
        this._videoHeartbeat.trackSeekComplete();
    };

    AnalyticsProvider.prototype._onComplete = function(e) {
        console.log('Player event: COMPLETE');
        this._videoHeartbeat.trackComplete();
    };

    // Export symbols.
    window.AnalyticsProvider = AnalyticsProvider;
})(window.ADB, window.Configuration, window.SamplePlayerDelegate);

File: sdks/android/samples/VideoHeartbeatSample/src/com/adobe/primetime/va/samples/analytics/AnalyticsProvider.java
package com.adobe.primetime.va.samples.analytics;

import android.util.Log;
import com.adobe.primetime.va.ConfigData;
import com.adobe.primetime.va.adb.VideoHeartbeat;
import com.adobe.primetime.va.samples.Configuration;
import com.adobe.primetime.va.samples.player.PlayerEvent;
import com.adobe.primetime.va.samples.player.VideoPlayer;

import java.util.Observable;
import java.util.Observer;

public class AnalyticsProvider implements Observer {
    private static final String LOG_TAG = "[VideoHeartbeatSample]::" + AnalyticsProvider.class.getSimpleName();

    private VideoPlayer _player;
    private SamplePlayerDelegate _playerDelegate;
    private VideoHeartbeat _videoHeartbeat;

    public void attachToPlayer(VideoPlayer player) {
        if (_player != null) {
            detachFromPlayer();
        }

        _player = player;
        _player.addObserver(this);

        _playerDelegate = new SamplePlayerDelegate(_player);

        ConfigData configData = new ConfigData(Configuration.HEARTBEAT_TRACKING_SERVER,
                                               Configuration.HEARTBEAT_JOB_ID,
                                               Configuration.HEARTBEAT_PUBLISHER);

        configData.ovp = Configuration.HEARTBEAT_OVP;
        configData.sdk = Configuration.HEARTBEAT_SDK;
        configData.channel = Configuration.HEARTBEAT_CHANNEL;

        // Set this to false for production apps.
        configData.debugLogging = true;

        _videoHeartbeat = new VideoHeartbeat(_playerDelegate);
        _videoHeartbeat.configure(configData);
    }

    public void detachFromPlayer() {
        if (_player != null) {
            _videoHeartbeat.destroy();
            _videoHeartbeat = null;
            _playerDelegate = null;

            _player.deleteObserver(this);
            _player = null;
        }
    }

    @Override
    public void update(Observable observable, Object data) {
        PlayerEvent playerEvent = (PlayerEvent)data;

        switch (playerEvent) {
            case VIDEO_LOAD:
                Log.d(LOG_TAG, "Video loaded.");
                _videoHeartbeat.trackVideoLoad();
                break;

            case VIDEO_UNLOAD:
                Log.d(LOG_TAG, "Video unloaded.");
                _videoHeartbeat.trackVideoUnload();
                break;

            case PLAY:
                Log.d(LOG_TAG, "Playback started.");
                _videoHeartbeat.trackPlay();
                break;

            case PAUSE:
                Log.d(LOG_TAG, "Playback paused.");
                _videoHeartbeat.trackPause();
                break;

            case SEEK_START:
                Log.d(LOG_TAG, "Seek started.");
                _videoHeartbeat.trackSeekStart();
                break;

            case SEEK_COMPLETE:
                Log.d(LOG_TAG, "Seek completed.");
                _videoHeartbeat.trackSeekComplete();
                break;

            case BUFFER_START:
                Log.d(LOG_TAG, "Buffer started.");
                _videoHeartbeat.trackBufferStart();
                break;

            case BUFFER_COMPLETE:
                Log.d(LOG_TAG, "Buffer completed.");
                _videoHeartbeat.trackBufferComplete();
                break;

            case COMPLETE:
                Log.d(LOG_TAG, "Playback completed.");
                _videoHeartbeat.trackComplete();
                break;

            default:
                Log.d(LOG_TAG, "Unhandled player event: " + playerEvent.toString());
                break;
        }
    }
}
